plasma_framework/test/endToEndTests/QuasarExit.e2e.test.js

```diff
@@ -1485,6 +1544,776 @@ contract(
+                                            );
+                                        });
+
+                                        it('should be able to withdraw all qTokens', async () => {
+                                            const bobQTokenBalanceAfterWithdraw = await this.qEth.balanceOf(bob);
+                                            const carolQTokenBalanceAfterWithdraw = await this.qEth.balanceOf(carol);
+
+                                            expect(bobQTokenBalanceAfterWithdraw).to.be.bignumber.equal(new BN(0));
+                                            expect(carolQTokenBalanceAfterWithdraw).to.be.bignumber.equal(new BN(0));
+                                        });
+
+                                        it('should update the quasar capacity and balance', async () => {
+                                            const quasarCapacityAfterWithdraw = new BN(
+                                                await this.quasar.tokenUsableCapacity(ETH),
+                                            );
+                                            const quasarBalance = new BN(
+                                                await web3.eth.getBalance(this.quasar.address),
+                                            );
+                                            const { poolSupply } = await this.quasar.tokenData(ETH);
+
+                                            expect(quasarCapacityAfterWithdraw).to.be.bignumber.equal(
+                                                quasarBalance,
+                                            ).equal(poolSupply);
+                                            expect(quasarBalance).to.be.bignumber.at.most(
+                                                new BN(10),
+                                            ).at.least(new BN(0));
+                                        });
+
+                                        it('should distribute returns in accordance with the supplied amount', async () => {
+                                            const bobBalanceAfterWithdraw = new BN(await web3.eth.getBalance(bob));
+                                            const carolBalanceAfterWithdraw = new BN(await web3.eth.getBalance(carol));
+
+                                            const bobReturnProfit = bobBalanceAfterWithdraw.add(
+                                                await spentOnGas(this.bobWithdrawTx.receipt),
+                                            ).sub(this.bobBalanceBeforeWithdraw).subn(this.bobSuppliedFunds);
+
+                                            const carolReturnProfit = carolBalanceAfterWithdraw.add(
+                                                await spentOnGas(this.carolWithdrawTx.receipt),
+                                            ).sub(this.carolBalanceBeforeWithdraw).subn(this.carolSuppliedFunds);
+
+                                            expect(carolReturnProfit).be.bignumber.above(bobReturnProfit);
```

Comment:
This seems a bit fuzzy. Is it not possible to calculate the exact expected profits?

Reply:
yeah 👍 I'll add that, will replicate the calculations here then, since we don't have a method yet that could withdraw the amount(in root tokens) you want to.